src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -256,6 +256,14 @@ typeSpec[boolean addImagNode]
     | builtInTypeSpec[addImagNode]
     ;
 
+// A type specification for a variable length parameter is a type name with
+// possible brackets afterwards that can end with annotations.
+variableLengthParameterTypeSpec
+    :   (classOrInterfaceType[false] | builtInType)
```

Comment:
This seems like a duplicate of `typeSpec`. Why doesn't what called this remain as `typeSpec` but add the annotation stuff afterwards?

Reply:
I was thinking the same thing, and I think that's what I tried first, basically using
```
typeSpec[false] ({LA(1) == AT}? annotations | )
```
to just add those optional annotations at the end, before `...`

But an example like `@Nullable String @Nullable ... vararg` fails to parse with "expecting `LBRACK`, found `...`" via
```
classTypeSpec[boolean addImagNode]
    :   classOrInterfaceType[addImagNode]
        (options{greedy=true; }:
            ({LA(1) == AT}? annotations
            | )
                lb:LBRACK^ {#lb.setType(ARRAY_DECLARATOR);} RBRACK)*
```
matching `String @Nullable` with `classOrInterfaceType annotations`.

So what I came up with as a workaround is almost a duplicate of `typeSpec`, but slightly different... taking
* `classOrInterfaceType (annotations? LBRACK RBRACK)* annotations?` and instead writing
* `classOrInterfaceType annotations? (LBRACK RBRACK annotations?)*`

where then it can parse `String @Nullable ...` and the other examples I added in the InputAnnotations12 file.

Does that sound reasonable? Or is there an issue, or better way to do it?

Reply:
I am fine with it, I am not seeing anything obvious now that I look at it closer.